### PR TITLE
JWT validator func

### PR DESCRIPTION
**Pull request by Alma-media**
Implements JWT validation for
- HMAC
- RSA

### CODE REVIEW

File: examples/jwt/main.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	dgrijalva "github.com/dgrijalva/jwt-go"
```
Comment (jeroenrinzema) [inline examples/jwt/main.go:4]: Maybe rename the alias for this package to: `jwtgo`?

File: pkg/functions/lib/jwt/jwt.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				Label: labels.Optional,
+			},
+		},
+		Type:  types.Message,
+		Label: labels.Required,
+	}
+}
+
+// JWT creates custom function to validate JWT with provided validator.
+func JWT(reader Reader, newClaims func() Claims) functions.Intermediate {
```
Comment (jeroenrinzema) [inline pkg/functions/lib/jwt/jwt.go:32]: Maybe rename this method to `New` to avoid stutter (`jwt.JWT`)?

File: pkg/functions/lib/jwt/errors.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"errors"
+	"fmt"
+
+	"github.com/jexia/semaphore/pkg/specs/types"
+)
+
+var (
+	errInvalidToken       = errors.New("token is invalid")
+	errInvalidValueType   = errors.New("invalid value, expected a string")
+	errMalformedAuthValue = errors.New("malformed auhtorization value")
```
Comment (jeroenrinzema) [inline pkg/functions/lib/jwt/errors.go:13]: These errors should be exported if we want other packages to perform type checks.

**Review by jeroenrinzema — CHANGES_REQUESTED** [review]
Great PR, I added a couple of comments. It would be useful to include the `JWT` function also inside the `daemon` that it could be used inside flows outside of the example.

File: pkg/functions/lib/jwt/errors.go (hunk 3 above)
Comment (Alma-media) [reply pkg/functions/lib/jwt/errors.go:13]: not sure it is needed
(they were introduced to perform error checks in tests)

**Review by jeroenrinzema — APPROVED** [review]

**Pull request merged**